design/work_item_board.go

```diff
@@ -0,0 +1,109 @@
+	workItemBoardLinks,
+)
+
+var workItemBoardList = JSONList(
+	"workItemBoard",
+	`List of work item boards`,
+	workItemBoardData,
+	workItemBoardLinks,
+	nil,
+)
+
+var workItemBoardLinks = a.Type("WorkItemBoardLinks", func() {
+	a.Attribute("self", d.String, func() {
+		a.Example("http://api.openshift.io/api/spacetemplates/2d98c73d-6969-4ea6-958a-812c832b6c18/workitemboards")
+	})
+	a.Required("self")
+})
+
+var workItemBoardColumnData = a.Type("WorkItemBoardColumnData", func() {
+	a.Description(`a column represents a vertical lane in a board`)
+	a.Attribute("type", d.String, "The type string of the work item board column", func() {
+		a.Enum("boardcolumns")
+	})
+	a.Attribute("id", d.UUID, "ID of the work item board column", func() {
+		a.Example("712f20e4-2202-4469-9a02-b892b7051b2b")
+	})
+	a.Attribute("attributes", workItemBoardColumnAttributes)
+	a.Required("id", "type", "attributes")
+})
+
+var workItemBoardColumnAttributes = a.Type("WorkItemBoardColumnAttributes", func() {
+	a.Attribute("name", d.String)
+	a.Attribute("order", d.Integer)
+	// TODO(michaelkleinhenz): as soon as we allow column customization, we need
+	// to also provide transRuleKey and transRuleArguments.
+	a.Required("name")
+})
+
+var workItemBoardData = a.Type("WorkItemBoardData", func() {
+	a.Description(`a board shows different work item type together in a board`)
+	a.Attribute("type", d.String, "The type string of the work item board", func() {
+		a.Enum("workitemboards")
+	})
+	a.Attribute("id", d.UUID, "ID of the work item board", func() {
+		a.Example("40bbdd3d-8b5d-4fd6-ac90-7236b669af04")
+	})
+	a.Attribute("attributes", workItemBoardAttributes)
+	a.Attribute("relationships", workItemBoardRelationships)
+	a.Attribute("links", genericLinks)
+	a.Required("type", "attributes")
+})
+
+var workItemBoardAttributes = a.Type("WorkItemBoardAttributes", func() {
+	a.Attribute("name", d.String)
+	a.Attribute("version", d.Integer, "Version for optimistic concurrency control")
+	a.Attribute("created-at", d.DateTime, "timestamp of entity creation")
+	a.Attribute("updated-at", d.DateTime, "timestamp of last entity update")
+	a.Attribute("context", d.String, "Context of this board")
+	a.Attribute("contextType", d.String, "Type of the context, used in addition to the context value")
+	a.Required("name", "context", "contextType")
+})
+
+var workItemBoardRelationships = a.Type("WorkItemBoardRelationships", func() {
+	a.Attribute("columns", relationGenericList, "List of work item board columns attached to the board")
+	a.Attribute("spaceTemplate", relationGeneric, "The space template to which this board belongs")
+})
+
+var _ = a.Resource("work_item_boards", func() {
+	a.BasePath("/workitemboards")
+	a.Parent("space_template")
+
+	a.Action("list", func() {
+		a.Routing(
+			a.GET(""),
+		)
+		a.Description("List work item boards")
+		a.Response(d.OK, workItemBoardList)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
```

Comment:
We could add `a.UseTrait("conditional")` here (https://github.com/fabric8-services/fabric8-wit/blob/master/design/spaces.go#L144)

Reply:
We sure should do this but apart from cache optimization this doesn't improve the stability of the system, would it? I mean it generates all the `NotModifiedSince` `NotMatching` tests but apart from that anything else?